File: week02/cpp/lab2.cpp
//
// Created by aonur on 6.03.2026.
//

#include "lab2.h"

#include <iostream>
#include <string>

using namespace std;

int main() {
    int x;
    int counter = 0;

    cout << "Enter a positive integer greater than 9: ";
    cin >> x;

    cout << x;


    while (x > 9) {
        int sum_digits = 0;
        int temp = x;


        while (temp > 0) {
            sum_digits += temp % 10;
            temp /= 10;
        }


        cout << " -> " << sum_digits;

        x = sum_digits;
        counter++;
    }

    cout << endl;
    cout << "Final value : " << x << endl;
    cout << "Total steps: " << counter << endl;







    int num;
    int fizzCount = 0;
    int buzzCount = 0;
    int fizzBuzzCount = 0;


    cout << "Please enter a number between 10 and 100: ";
    cin >> num;

    while (num < 10 || num > 100) {
        cout << "Invalid input. Please enter a number between 10 and 100: ";
        cin >> num;
    }


    for (int i = 1; i <= num; i++) {

        if (i % 7 == 0) {
            cout << "(" << i << " is skipped)" << endl;
            continue;
        }


        if (i % 3 == 0 && i % 5 == 0) {
            cout << "FizzBuzz" << endl;
            fizzBuzzCount++;
        }

        else if (i % 3 == 0) {
            cout << "Fizz" << endl;
            fizzCount++;
        }

        else if (i % 5 == 0) {
            cout << "Buzz" << endl;
            buzzCount++;
        }

        else {
            cout << i << endl;
        }
    }


    cout << "\n--Summary--" << endl;
    cout << "Fizz count     : " << fizzCount << endl;
    cout << "Buzz count     : " << buzzCount << endl;
    cout << "FizzBuzz count : " << fizzBuzzCount << endl;







    int n;


    cout << "Please enter a number between 3 and 9: ";
    cin >> n;


    if (n >= 3 && n <= 9) {


        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= i; j++) {
                cout << j;
            }
            cout << endl;
        }


        for (int i = n - 1; i >= 1; i--) {
            for (int j = 1; j <= i; j++) {
                cout << j;
            }
            cout << endl;
        }

    } else {
        cout << "Invalid input!" << endl;
    }

    return 0;
}

File: week02/python/lab2.py
x=int(input("Enter a number greater than 9"))

counter=0
print(x, end="")
while x >9:
    sum=0
    temp=x

    while x>0:
        sum=sum + x%10
        x = x // 10

    print(f" -> {sum}", end="")

    counter=counter+1
    x = sum
print()
print("Final value :" , sum)
print("Total steps:", counter)



x=int(input("Enter a number between 10 and 100 inclusive:"))
while x<10 or x>100:
     x=int(input("Invalid input.Enter a number between 10 and 100 inclusive:"))

i=1
fizcount=0
buzcount=0
fbcount=0
while i<= x:
     if i%7==0:
         print(i,"is skipped")
     elif i%15 ==0:
         print("FizzBuzz")
         fbcount+=1

     elif i%3 ==0:
         print("Fizz")
         fizcount+=1

     elif i%5 ==0:
         print("Buzz")
         buzcount+=1
     else :
         print(i)
     i +=1


print("--Summary--")
print("Fizz count : ", fizcount)
print("Buzz count : ", buzcount)
print("FizBuzz count : ", fbcount)

n = int(input("Please enter a number between 3 and 9: "))
if 3 <= n <= 9:

    for i in range(1, n + 1):
        for j in range(1, i + 1):
            print(j, end="")
        print()


    for i in range(n - 1, 0, -1):
        for j in range(1, i + 1):
            print(j, end="")
        print()
else:
    print("Invalid input!")